packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -128,10 +130,11 @@ function NavigationMenuSelector( {
 	] );
 
 	const NavigationMenuSelectorDropdown = (
-		<DropdownMenu
+		<ToolbarDropdownMenu
 			label={ selectorLabel }
-			icon={ moreVertical }
-			toggleProps={ { isSmall: true } }
+			icon={ icon }
+			toggleProps={ toggleProps }
```

Comment:
Augmenting toggle props with `aria*` attribute to address Alex's feedback re: text label content could work here.